### DISCUSSION TITLE
Interpolation of Resample class

### DISCUSSION THREAD

**Discussion by kodohoi** [TorchIO-project/torchio#1085] [Q&A] [answered]
When I used `Resample` class to change the spacial infromation of CT images, my CT images looked distorted on 3D view in 3D slicer. Is this problem caused by linear interpolation which is the default interpolation of Torchio? I confirmed the affine of CT images changed after transformation of Resample.

**Comment by fepegar** [comment]
Can you explain what you run, what you expected and share a screenshot of what you got?

**Reply by kodohoi** [reply]
The original image size was (512,512,512). However, this was too large to train the model. I used `Resample` and `CropOrPad` class to resize the images smaller. Here is the code I implemented.

```
def remove_zero_rows_columns_torchio(image):  #function to remove the rows with 0.
      img = image.tensor.data[0]
      row_sum = img.sum(dim=(1, 2)) > 0
      col_sum = img.sum(dim=(0, 2)) > 0
      slice_sum = img.sum(dim=(0, 1)) > 0

      reduced_tensor = image.tensor[:, row_sum, :,:][:,:, col_sum,:][:,:, :, slice_sum]

      reduced_image = tio.ScalarImage(tensor=reduced_tensor, affine=image.affine)
      return reduced_image

for patient in patients_list:
      id = patient.split("_")[1].split("-")[0]
      patient_path = os.path.join(path, patient)
      image = tio.ScalarImage(patient_path)
      image.data = torch.where(image.data < 500, 0.0, 1.0)  #Binarization
      transforms = [
            tio.Resample(1),  #downsampling
            tio.ToCanonical(),
            tio.CropOrPad((300,300,300)),
        ]
      image = remove_zero_rows_columns_torchio(image)
      transform = tio.Compose(transforms)
      image = transform(image)
      image.save(os.path.join("./resized_directiry", id + ".nii.gz"))
```

After this preprocessing, the images looks distorted partially on 3D slicer as below.

![画像](https://github.com/fepegar/torchio/assets/106305375/6074f42b-03b9-43fe-a7d7-42472ac83b69)
I guess `ToCanonical` and `CropOrPad` class don't affect the spacing information. 
When I asked ChatGPT, it said "The main factor that can potentially lead to distortions is the choice of the interpolation method.".
Is my implementation correct? If so, I will consider another possibility.

**Reply by fepegar** [reply]
Thanks. Can you please share a minimal example and one image to try it on? Also, it might be more useful to show us the 2D views in Slicer, instead if a posterior 3D view. I think ChatGPT lacks some context in this case. It might think you're referring to intensity distortion, as opposed to geometric distortions.

A bit off topic, but you could instead open the PDF of the TorchIO documentation (https://torchio.readthedocs.io/_/downloads/en/latest/pdf/) in Edge, ask the Bing Chat to read it, and ask questions.

**Reply by kodohoi** [reply]
Thank you.
[ZA-torchio_000.nii.gz](https://github.com/fepegar/torchio/files/11581164/ZA-torchio_000.nii.gz)
[ZA-torchio_000.zip](https://github.com/fepegar/torchio/files/11581165/ZA-torchio_000.zip)
I share the DICOM images before processing and processed nifti image.
![image2](https://github.com/fepegar/torchio/assets/106305375/afc31c5f-7775-4453-9438-68e7506a64d5)

**Reply by kodohoi** [reply]
@fepegar 
I apologize for the inconvinience. 
I think this geometoric distortion of my data may be caused by Gantry/Detector Tilt.
I guess nifti files do not contain the information about the tilt angle of a CT, and this result in the geometirc distortion...
I am very sorry...

Does the tilt angle affect the affine matrix of image?
If so, is it possible to deal with the effect of the tilt angle with Torchio?

**Reply by fepegar** [reply, marked as answer]
No worries. Yes, the tilt angle will affect the affine. But it's probably stored in the NIfTI sform matrix, because the qform can only store a rigid transformation. You could resample your image using the sform so the tilting is not visible in your voxel data.

**Reply by fepegar** [reply]
I recommend you fix your images as explained in the Slicer forum: https://discourse.slicer.org/t/gantry-tilt-correction-in-ct-dicoms/19443
